packages/core/core/src/requests/WriteBundleRequest.js

```diff
@@ -66,11 +66,12 @@ export type WriteBundleRequest = {|
 export default function createWriteBundleRequest(
   input: WriteBundleRequestInput,
 ): WriteBundleRequest {
+  let name = nullthrows(input.bundle.name);
   let nameHash = nullthrows(
     input.hashRefToNameHash.get(input.bundle.hashReference),
   );
   return {
-    id: `${input.bundle.id}:${input.info.hash}:${nameHash}`,
+    id: `${input.bundle.id}:${input.info.hash}:${nameHash}:${name}`,
```

Comment:
What was the reason for this change?  When does a bundle change name where its hash doesn't change?

Reply:
This happens for bundles that have their `needsStableName` field changed.